crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -291,97 +291,93 @@ pub fn handle_runnables(
+            res.push("--".to_string());
+            res.push(name.to_string());
+            res.push("--nocapture".to_string());
+        }
+        RunnableKind::TestMod { path } => {
+            res.push("test".to_string());
+            if let Some(spec) = spec {
+                spec.push_to(&mut res);
             }
-            RunnableKind::Bin => {
-                res.push("run".to_string());
-                if let Some(spec) = spec {
-                    spec.push_to(&mut res);
-                }
+            res.push("--".to_string());
+            res.push(path.to_string());
+            res.push("--nocapture".to_string());
+        }
+        RunnableKind::Bin => {
+            res.push("run".to_string());
+            if let Some(spec) = spec {
+                spec.push_to(&mut res);
             }
         }
-        Ok(res)
     }
+    Ok(res)
+}
 
-    struct CargoTargetSpec {
-        package: String,
-        target: String,
-        target_kind: TargetKind,
-    }
+struct CargoTargetSpec {
+    package: String,
+    target: String,
+    target_kind: TargetKind,
+}
 
-    impl CargoTargetSpec {
-        fn for_file(world: &ServerWorld, file_id: FileId) -> Result<Option<CargoTargetSpec>> {
-            let &crate_id = match world.analysis().crate_for(file_id)?.first() {
-                Some(crate_id) => crate_id,
-                None => return Ok(None),
+impl CargoTargetSpec {
+    fn for_file(world: &ServerWorld, file_id: FileId) -> Result<Option<CargoTargetSpec>> {
+        let &crate_id = match world.analysis().crate_for(file_id)?.first() {
+            Some(crate_id) => crate_id,
+            None => return Ok(None),
+        };
+        let file_id = world.analysis().crate_root(crate_id)?;
+        let path = world
+            .vfs
+            .read()
+            .file2path(ra_vfs::VfsFile(file_id.0.into()));
+        let res = world.workspaces.iter().find_map(|ws| {
+            let tgt = ws.cargo.target_by_root(&path)?;
+            let res = CargoTargetSpec {
+                package: tgt.package(&ws.cargo).name(&ws.cargo).to_string(),
+                target: tgt.name(&ws.cargo).to_string(),
+                target_kind: tgt.kind(&ws.cargo),
             };
-            let file_id = world.analysis().crate_root(crate_id)?;
-            let path = world
-                .vfs
-                .read()
-                .file2path(ra_vfs::VfsFile(file_id.0.into()));
-            let res = world.workspaces.iter().find_map(|ws| {
-                let tgt = ws.cargo.target_by_root(&path)?;
-                let res = CargoTargetSpec {
-                    package: tgt.package(&ws.cargo).name(&ws.cargo).to_string(),
-                    target: tgt.name(&ws.cargo).to_string(),
-                    target_kind: tgt.kind(&ws.cargo),
-                };
-                Some(res)
-            });
-            Ok(res)
-        }
+            Some(res)
+        });
+        Ok(res)
+    }
```

Comment:
Let's move `CargoTargetSpec` and `runnable_args` to a separate top-level module in `ra-lsp-server`?

Reply:
Done